plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -186,8 +186,8 @@ public ConnectorPageSource createPageSource(
         Schema requestedSchema = constructSchema(dataSchema, columnHandles.stream().map(HiveColumnHandle::getName).toList());
 
         TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-        // Only enable predicate pushdown for COW tables
-        boolean enablePredicatePushDown = ((HudiTableHandle) connectorTable).getTableType().equals(HoodieTableType.COPY_ON_WRITE);
+        // Enable predicate pushdown for splits containing only base files
+        boolean isBaseFileOnly = ((HudiTableHandle) connectorTable).getTableType().equals(HoodieTableType.COPY_ON_WRITE) || hudiSplit.getLogFiles().isEmpty();
```

Comment:
Could we remove table type check `(HudiTableHandle) connectorTable).getTableType().equals(HoodieTableType.COPY_ON_WRITE)` and completely rely on the log file check?

Reply:
Done